hudi-common/src/main/java/org/apache/parquet/schema/AvroSchemaRepair.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.parquet.schema;
+
+import org.apache.avro.LogicalType;
+import org.apache.avro.LogicalTypes;
+import org.apache.avro.Schema;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AvroSchemaRepair {
+  public static Schema repairLogicalTypes(Schema requestedSchema, Schema tableSchema) {
+    if (requestedSchema.getType() != Schema.Type.RECORD) {
+      return requestedSchema;
+    }
+
+    List<Schema.Field> repairedFields = new ArrayList<>();
+
+    for (Schema.Field requestedField : requestedSchema.getFields()) {
+      Schema.Field repaired = requestedField;
+      Schema.Field tableField = tableSchema.getField(requestedField.name());
+      if (tableField != null) {
+        repaired = repairAvroField(requestedField, tableField);
+      }
+      repairedFields.add(repaired);
+    }
+
+    return Schema.createRecord(
```

Comment:
1. there is no need to instantiate a new schema if the timestmap type matches in precision;
2. we better intern the schema if is new generated